### PR TITLE
Contributable variables for configuration resolver

### LINKED ISSUE #32239: Run the default build task from launch configuration
Currently, I can specify a `preLaunchTask` field in a launch configuration to run a task before launching. That's great! To use preLaunchTask, I need to give my task an identifier first. Typically, I want to run the default build task before I launch, though. VSCode already knows which task is the default build task, so I feel like there should be a way to tell VSCode to run this task without giving it an identifier.

If one of my tasks.json entries contains the following snippet, VSCode knows that this is the default build task:

```
"group": {
	"kind": "build",
	"isDefault": true
}
```

What if there were a field in launch.json named something like `preLaunchBuild` that could be set to `true` to tell it to run the default build task before launching? `preLaunchTask` would still be useful for running other tasks, but if I just want to run the default build task, I don't need to take the extra step of giving the default build task its own identifier.

### PR DESCRIPTION
**Pull request by alexr00**
Fixes https://github.com/microsoft/vscode/issues/32239

### CODE REVIEW

**Comment by alexr00** [comment]
@isidorn I've added a way for other parts to contribute variables. Can you take a look at https://github.com/microsoft/vscode/pull/80478/files#diff-db37190de07b5eba73c934622c1ab1f5 in particular and let me know if you think it makes sense?

File: src/vs/workbench/services/configurationResolver/common/configurationResolver.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ export interface IConfigurationResolverService {
 	 */
 	resolveAny(folder: IWorkspaceFolder | undefined, config: any, commandValueMapping?: IStringDictionary<string>): any;
 
+	/**
+	 * Like resolveAny, but also includes variable contributed by other parts of the system.
+	 */
+	resolveAnyIncludeContributed(folder: IWorkspaceFolder | undefined, config: any, commandValueMapping?: IStringDictionary<string>): Promise<any>;
```
Comment (isidorn) [inline src/vs/workbench/services/configurationResolver/common/configurationResolver.ts]: Why do we have a special `resolveAnyIncludeContributed`?
I think `resolveAny` shuold always take care of contributed variables

**Comment by isidorn** [comment]
There seems to be more changes here than what is simply needed for contribuatble varialbes it seems to me? So I did not give a deep reivew of all the changes,

Also it would be great if you write some unit tests that cover this - they first contribute a variable and then make sure it gets nicely resolved.

**Comment by alexr00** [comment]
@isidorn thank you for reviewing the internal API surface! I have consolidated to just `resolveAny` and added a test.

**Comment by isidorn** [comment]
Great. It looks good to me now, feel free to merge.

**Pull request merged**